src/foam/java/refinements.js

```diff
@@ -321,6 +321,33 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  refines: 'foam.core.Action',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'javaCode'
+    },
+    {
+      class: 'String',
+      name: 'javaReturns'
```

Comment:
This was added in because I was not sure whether the javaCode would allow for methods to be created that would return a specific type instead of being void all the time.